webnovel/bot.py

```diff
@@ -116,10 +116,13 @@ def signin(self, email, password):
         WebDriverWait(self.driver, self.timeout).until(
             EC.presence_of_element_located((By.CSS_SELECTOR, "a[title='My Profile']"))
         )
+        # click the preferences button away
+        signin_btn = self.driver.find_element_by_class_name('j_post_preference') # bt _m mw160 j_post_preference
```

Comment:
preference popup seems to take sometime to load, so we should preferably wait and check for a time

```python
WebDriverWait(self.driver, 60).until(
    EC.presence_of_element_located((By.CSS_SELECTOR, ".j_post_preference"))
)
```

this throws a `TimeoutException` when timer has run out without finding the element. at which point we can just ignore the preference popup

___

the current button selection does not seem to work

```python
self.driver.find_element_by_css_selector('.j_post_preference button')
```

above code selects the first button that appear under preference panel

Reply:
>the current button selection does not seem to work

It works for me. Your code 

```python
# wait for the preferences popup to load and click it away
WebDriverWait(self.driver, 60).until(
  EC.presence_of_element_located((By.CSS_SELECTOR, ".j_post_preference"))
)
signin_btn = self.driver.find_element_by_css_selector('.j_post_preference button') # bt _m mw160 j_post_preference
signin_btn.click()
```
doesn't.

I suspect A/B testing. Are we getting the same page code?

![image](https://user-images.githubusercontent.com/1641362/108879607-b6de5100-7601-11eb-9559-dbe040baa429.png)

Reply:
EDIT: Actually it doesn't work for me either now, huh.

Reply:
I added the wait which indeed seemed to be an issue and used the shorter click function in the other place.

Reply:
indeed, my bad. I mistyped. i meant.

```python
self.driver.find_element_by_css_selector('.j_preference_form button')
```

preference form doesnt appear all the time and without an exception catch some cases wont work.

But, this seems to work so i'll merge and add that later